test/end2end_test.go

```diff
@@ -5509,6 +5510,106 @@ func (s) TestClientForwardsGrpcAcceptEncodingHeader(t *testing.T) {
 	}
 }
 
+func (s) TestServerHandlerSetSendCompression(t *testing.T) {
```

Comment:
Could we actually split this test into multiple ones to simplify the test logic. Please see: https://google.github.io/styleguide/go/decisions#table-driven-tests